handlers_test.go

```diff
@@ -0,0 +1,414 @@
+package firecracker
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestHandlerListAppend(t *testing.T) {
+	h := HandlerList{}
+	h.Append(Handler{Name: "foo"})
+
+	if size := h.Len(); size != 0 {
+		t.Errorf("expected length to be '0', but received '%d'", size)
+	}
+
+	expectedNames := []string{
+		"foo",
+		"bar",
+		"baz",
+	}
+
+	for _, name := range expectedNames {
+		h = h.Append(Handler{Name: name})
+	}
+
+	for i, name := range expectedNames {
+		if e, a := name, h.list[i].Name; e != a {
+			t.Errorf("expected %q, but received %q", e, a)
+		}
+	}
+}
+
+func TestHandlerListRemove(t *testing.T) {
+	h := HandlerList{}.Append(
+		Handler{
+			Name: "foo",
+		},
+		Handler{
+			Name: "bar",
+		},
+		Handler{
+			Name: "baz",
+		},
+		Handler{
+			Name: "foo",
+		},
+		Handler{
+			Name: "baz",
+		},
+	)
+
+	h.Remove("foo")
+
+	if e, a := 5, h.Len(); e != a {
```

Comment:
Is it worth also testing that the index we expected to be remove was the one removed? like
```go
if h[0] == "foo" {
        t.Errorf("expected foo to be removed but was not")
}
```